manifests/install.pp

```diff
@@ -61,9 +64,16 @@
         source            => $::puppet_agent::source,
       }
     }
+  } elsif ($::osfamily == 'Solaris' and $::operatingsystemmajrelease == '10') or $::osfamily == 'Darwin' {
+    # Solaris 10/OSX package provider does not provide 'versionable'
+    # Package is removed above, then re-added as the new version here.
+    package { $::puppet_agent::package_name:
+      ensure => 'present',
```

Comment:
Yeah, since Solaris doesn't support versioned, we can't give a version here. So we just ensure that the source is the correct version.